src/uct/tcp/tcp_sockcm_ep.c

```diff
@@ -862,12 +942,47 @@ static ucs_status_t uct_tcp_sockcm_ep_server_create(uct_tcp_sockcm_ep_t *tcp_ep,
               tcp_ep->fd, params_tcp_sockcm, tcp_ep->state);
 
     /* now that the server's ep was created, can try to send data */
-    ucs_async_modify_handler(tcp_ep->fd, UCS_EVENT_SET_EVWRITE | UCS_EVENT_SET_EVREAD);
-    return UCS_OK;
+    ucs_async_modify_handler(tcp_ep->fd, UCS_EVENT_SET_EVWRITE |
+                                         UCS_EVENT_SET_EVREAD);
+
+    if (ucs_test_all_flags(params->field_mask,
+                           UCT_EP_PARAM_FIELD_PRIV_DATA |
+                           UCT_EP_PARAM_FIELD_PRIV_DATA_LENGTH)) {
+        priv_data        = params->private_data;
+        priv_data_length = params->private_data_length;
+    } else if (params->field_mask & UCT_EP_PARAM_FIELD_SOCKADDR_PACK_CB) {
+        status = ucs_sockaddr_get_ifname(tcp_ep->fd, ifname_str,
+                                         sizeof(ifname_str));
+        if (UCS_OK != status) {
+            goto err_unblock;
+        }
+
+        pack_args.field_mask = UCT_CM_EP_PRIV_DATA_PACK_ARGS_FIELD_DEVICE_NAME;
+        ucs_strncpy_safe(pack_args.dev_name, ifname_str, UCT_DEVICE_NAME_MAX);
+        data_buf = ucs_malloc(tcp_sockcm->priv_data_len, "tcp_priv_data");
+        status   = uct_cm_ep_pack_cb(&tcp_ep->super, tcp_ep->super.user_data,
+                                     &pack_args, data_buf,
+                                     tcp_sockcm->priv_data_len,
+                                     &priv_data_ret);
+        if (status != UCS_OK) {
+            tcp_ep->state |= UCT_TCP_SOCKCM_EP_PACK_CB_FAILED;
+            goto err_unblock;
+        }
+
+        priv_data        = data_buf;
+        priv_data_length = priv_data_ret;
```

Comment:
maybe make helper function for this

Reply:
then need to handle allocated `data_buf` outside, that also does not look so good to me

Reply:
then maybe 
{
data = malloc
call-helper-function (including ucs_sockaddr_get_ifname, pack_cb)
}